apps/wrf/readme.md

```diff
@@ -1,110 +1,461 @@
+- Installs CycleCloud environment from scratch
+- Creates NFS storage server using CycleCloud cluster template
+- Installs WRF/WPS v4 software (via “azurehpc” scripts)
+- Submit jobs to run WRF v4 application for testing
+  
 ## Prerequisites
 
-Cluster is built with the desired configuration for networking, storage, compute etc. You can see the tutorial or examples folder in this repo for how to set this up. Spack is installed (See [here](../spack/readme.md) for details).
+- As this procedure uses HBv2 VMs to run WRFv4 simulations, you may need to request quota increase for this type of SKU in the subscription and region you will deploy the environment. 
 
-Dependencies for binary version:
 
-* None
+## Install Azure CycleCloud
 
+Follow the steps to [Install and Setup CycleCloud](../../tutorials/cyclecloud/install-cyclecloud.md)
+ 
+ 
+## Create NFS Storage cluster
+-	It can be possible to include an external NFS share at this point (in the example, I have shared from an NFS cluster using on CycleCloud template)
 
-First copy the apps directory to the cluster in a shared directory.  The `azhpc-scp` can be used to do this:
+![NFS-Cluster1](images/NFS-Cluster1.png)
+![NFS-Cluster2](images/NFS-Cluster2.png)
+![NFS-Cluster3](images/NFS-Cluster3.png)
 
+Changes:
+- Change OS to use CentOS 7 versions
+- Use +300GB storage size (space to download WRF data)
+- Change cloud-init as following:
+- Confirm the IP address of your NFS storage and change it below accordingly:
+
+```
+#!/bin/bash
+
+set -x
+yum install -y epel-release
+yum install -y Lmod at
+systemctl enable --now atd.service
+cat <<EOF>/mnt/exportfs.sh
+#!/bin/bash
+set -x
+mkdir -p /mnt/exports/data /mnt/exports/apps
+sudo exportfs -o rw,sync,no_root_squash 10.4.0.0/20:/mnt/exports/data
+sudo exportfs -o rw,sync,no_root_squash 10.4.0.0/20:/mnt/exports/apps
+EOF
+chmod 755 /mnt/exportfs.sh
+at now + 2 minute -f /mnt/exportfs.sh
 ```
-azhpc-scp -r $azhpc_dir/apps/. hpcuser@headnode:.
+
+Connect to NFS storage cluster and check the mounts are correct:
+```
+# check mount
+sudo exportfs -s
 ```
 
-> Alternatively you can checkout the azurehpc repository but you will need to update the paths according to where you put it.
+## Setup WRF cluster using HBv2 VM
+Summary of steps:
+-	Start NFS storage cluster on CycleCloud
+-	Import WRF cluster template (from “azurehpc” scripts) 
+-	Start WRF cluster using HBv2 VM 
+-	Install WRF/WPS 4 software (from “azurehpc” scripts) 
+-	Download data for WRF 4
+-	Edit data locations in WRF config files
+-	Generate WRF 4 input files, change permissions
+-	Run WRF 4  applications for testing
 
 
-## Connect to the headnode
+### Import custom CycleCloud template for WRF
 
+After start NFS storage cluster, ssh to it and download azurehpc GitHub repository
 ```
-azhpc-connect -u hpcuser headnode
+## Download azurehpc GitHub repository
+cd /data
+#git clone https://github.com/Azure/azurehpc.git
+git clone https://github.com/marcusgaspar/azurehpc.git
```

Comment:
Is this URL correct (you are pointing to your fork?)

Reply:
It is temporary, as I'm currently using my fork during POCs.